src/dynadojo/systems/gilpin_flows.py

```diff
@@ -0,0 +1,172 @@
+import numpy as np
+from ..abstractions import AbstractSystem
+import importlib
+import warnings
+import os
+import json
+import dysts
+from dysts.utils import generate_ic_ensemble
+
+class GilpinFlowsSystem(AbstractSystem):
+    """
+    Gilpin Flows System for modeling chaotic attractors using systems from the dysts library.
+
+    Attributes
+    ----------
+    base_path : str
+        Base path for the dysts library.
+    json_file_path : str
+        Path to the JSON file containing chaotic attractors data.
+
+    Example
+    -------
+    >>> from dynadojo.systems.gilpin_flows import GilpinFlowsSystem
+    >>> latent_dim = 3
+    >>> embed_dim = 3
+    >>> system_name = 'Lorenz'
+    >>> system = GilpinFlowsSystem(latent_dim, embed_dim, system_name)
+    >>> n = 10
+    >>> init_conds = system.make_init_conds(n, in_dist=True)
+    >>> timesteps = 100
+    >>> trajectories = system.make_data(init_conds, timesteps)
+    >>> error = system.calc_error(trajectories[0], trajectories[1])
+    >>> control_cost = system.calc_control_cost(np.zeros((n, timesteps, embed_dim)))
+
+    Methods
+    -------
+    __init__(self, latent_dim, embed_dim, system_name: str, seed=None)
+        Initializes the system with given dimensions and system name.
+    make_init_conds(self, n: int, in_dist=True) -> np.ndarray
+        Generates initial conditions for the system.
+    make_data(self, init_conds: np.ndarray, timesteps: int, control=None, noisy=False) -> np.ndarray
+        Generates trajectories from initial conditions.
+    calc_error(self, x: np.ndarray, y: np.ndarray) -> float
+        Calculates the mean squared error between two arrays.
+    calc_control_cost(self, control: np.ndarray) -> float
+        Calculates the control cost.
+    _all_systems(cls) -> list
+        Class method that loads systems data and returns the list of available systems, excluding missing systems.
+    """
+    base_path = os.path.dirname(dysts.__file__)
+    json_file_path = os.path.join(base_path, 'data', 'chaotic_attractors.json')
+
+    @classmethod
+    def _all_systems(cls):
+        """Load systems data and return the list of ll available systems."""
+        with open(cls.json_file_path, 'r') as file:
+            systems_data = json.load(file)
+
+        system_list = list(systems_data.keys())
+        
+        module = importlib.import_module('dysts.flows')
+        all_systems = [system_name for system_name in system_list if hasattr(module, system_name)]
+        
+        return all_systems
+
+    def __init__(self, latent_dim, embed_dim, system_name: str, pts_per_period=100, seed=None):
```

Comment:
Arguments should all be typed (like you did for system_name). See other systems for example.

Also need to give all params a default value.